src/vaadin-text-field-mixin.html

```diff
@@ -502,6 +503,18 @@
       }
     }
 
+    __constraintsChanged(required, minlength, maxlength, pattern, min, max, step) {
+      if (!this.invalid) {
+        return;
+      }
+
+      if (!required && !minlength && !maxlength && !pattern && !min && !max) {
```

Comment:
Oh, `min` and `max` can be set to 0 which is falsy value, but still a valid constraint. We should probably check for `undefined`/`null` here instead.

Reply:
Good catch. I'm sure I was thinking about this but forgot it while changing other things.

Atm it makes an invalid field valid if e.g. `max` is set to `0` and value is bigger than `0`. I'll add test cases for these min/max 0 validity changes and fix this.

Reply:
Fixing this revealed that the test `should update "invalid" state when "step" is removed` was a false positive before (it started failing). Fixed that test too and pushed an update.

Reply:
Or well the test was correct, but was passing due to broken behaviour. Fixing that behaviour just revealed that another fix was needed too.